geode-core/src/distributedTest/java/org/apache/geode/cache/ConnectionPoolDUnitTest.java

```diff
@@ -4585,85 +3250,59 @@ public void test029EmptyByteArray() throws CacheException {
+      getLonerSystem();
+      getCache();
+      RegionFactory<Object, Object> factory = getCache().createRegionFactory();
+      factory.setScope(Scope.LOCAL);
+      factory.setConcurrencyChecksEnabled(false);
 
-    vm1.invoke(create);
-    vm1.invoke("Create empty byte array", new CacheSerializableRunnable() {
-      @Override
-      public void run2() throws CacheException {
-        Region<Object, Object> region = getRootRegion().getSubregion(name);
-        for (int i = 0; i < 1; i++) {
-          region.create(i, new byte[0], createCallbackArg);
-        }
-      }
+      configureConnectionPool(factory, host0, new int[] {port}, true, -1, -1, null);
+      createRegion(name, factory);
     });
 
-    vm1.invoke("Verify values on client", new CacheSerializableRunnable() {
-      @Override
-      public void run2() throws CacheException {
-        Region<Object, Object> region = getRootRegion().getSubregion(name);
-        for (int i = 0; i < 1; i++) {
-          Region.Entry entry = region.getEntry(i);
-          assertNotNull(entry);
-          byte[] value = (byte[]) entry.getValue();
-          assertNotNull(value);
-          assertEquals(0, value.length);
-        }
+    vm1.invoke("Create empty byte array", () -> {
+      Region<Object, Object> region = getRootRegion().getSubregion(name);
+      for (int i = 0; i < 1; i++) {
+        region.create(i, new byte[0], createCallbackArg);
       }
     });
-    vm0.invoke("Verify values on server", new CacheSerializableRunnable() {
-      @Override
-      public void run2() throws CacheException {
-        Region<Object, Object> region = getRootRegion().getSubregion(name);
-        for (int i = 0; i < 1; i++) {
-          Region.Entry entry = region.getEntry(i);
-          assertNotNull(entry);
-          byte[] value = (byte[]) entry.getValue();
-          assertNotNull(value);
-          assertEquals(0, value.length);
-        }
+
+    vm1.invoke("Verify values on client", () -> {
+      Region<Object, Object> region = getRootRegion().getSubregion(name);
+      for (int i = 0; i < 1; i++) {
+        Region.Entry entry = region.getEntry(i);
+        assertThat(entry).isNotNull();
+        byte[] value = (byte[]) entry.getValue();
+        assertThat(value).isNotNull();
+        assertThat(0).isEqualTo(value.length);
       }
     });
 
-    SerializableRunnable close = new CacheSerializableRunnable("Close Pool") {
-      @Override
-      public void run2() throws CacheException {
-        Region<Object, Object> region = getRootRegion().getSubregion(name);
-        region.localDestroyRegion();
+    vm0.invoke("Verify values on server", () -> {
+      Region<Object, Object> region = getRootRegion().getSubregion(name);
+      for (int i = 0; i < 1; i++) {
+        Region.Entry entry = region.getEntry(i);
+        assertThat(entry).isNotNull();
+        byte[] value = (byte[]) entry.getValue();
+        assertThat(value).isNotNull();
+        assertThat(0).isEqualTo(value.length);
```

Comment:
These length or size related assertions are better done as:
```
assertThat(value).isEmpty();
```
or for non-empty:
```
assertThat(value).hasSize(2);
```
But don't ever use a PartitionedRegion as the value in the assertion because it will walk the map to create the error message and PR can actually hang trying to create buckets.